### ISSUE TITLE
Ruby client for remote driver is slow

### ISSUE THREAD

**Issue by jumpalongjim** [SeleniumHQ/selenium#1370]
There seems to be a 1 second delay on each command to a remote driver from Ruby. Using the same remote driver and test script, running from a .NET script took 9 seconds and from a Ruby script took 33 seconds. The remote driver log shows that command responses were delivered in typically 0.2 seconds in each case.

Is there some wait or delay built into Ruby? I can't see anything mentioned in the docs or source... It seems unlikely that Ruby itself could be so slow, and the delay of almost exactly 1 second per command looks too regular to be caused by language performance.

**Comment by titusfortner** [comment]
No, there is no wait or delay built into Ruby.
Are command responses 0.2 seconds or 1 second?
If you are comparing with .NET, I'm assuming this is windows? What browser are you testing with?

**Comment by jumpalongjim** [comment]
Thanks for your response. The remote driver in this case is [Winium](https://github.com/2gis/Winium), driving a desktop application. The timings were measured by the log output at the remote end, showing the http connections opening and closing generally in less than 0.1 seconds, interspersed with about 1 second pause until the next connection was requested. The client side ran on the same machine, firstly in Ruby. I observed the slow speed of the test (33 seconds) and started investigating. When I wrote the same script in .NET, calling in to the same remote driver with the same commands, the test took 9 seconds.

If there are no delays built in then I guess it's some configuration or fault at a different level on my machine.
